packages/davinci-client/src/lib/davinci.api.ts

```diff
@@ -79,45 +84,44 @@ export const davinciApi = createApi({
        * parameters are pre-typed from the library.
        */
       async onQueryStarted(_, api) {
+        let response;
+
         try {
-          await api.queryFulfilled;
-        } catch (error) {
+          const query = await api.queryFulfilled;
+          response = query.meta?.response;
+        } catch (err: unknown) {
+          const error = err as ThrownQueryError;
+
           /**
            * This error is thrown when the query is rejected. We don't
            * want to do anything with it for now.
            */
+          response = error.meta?.response;
         }
 
-        /**
-         * The original DaVinci response is appended to the cache, so we are going
-         * to pull it and dispatch the appropriate action based on the response.
-         */
-        const cacheEntry = api.getCacheEntry();
+        const cacheEntry: DaVinciCacheEntry = api.getCacheEntry();
 
-        /**
-         * Detect the type of response and dispatch the appropriate action
-         * This leads to a new node in the state
-         */
-        if (cacheEntry.isSuccess && 'eventName' in cacheEntry.data) {
-          const cacheNextEntry = cacheEntry as DaVinciCacheEntry<DavinciNextResponse>;
-          api.dispatch(nodeSlice.actions.next(cacheNextEntry));
-        } else if (cacheEntry.isSuccess && cacheEntry.data.status === 'COMPLETED') {
-          const cacheSuccessEntry = cacheEntry as DaVinciCacheEntry<DavinciSuccessResponse>;
-          api.dispatch(nodeSlice.actions.success(cacheSuccessEntry));
-        } else if (cacheEntry.isError) {
-          const cacheErrorEntry = cacheEntry as DaVinciErrorCacheEntry<DavinciErrorResponse>;
-          api.dispatch(nodeSlice.actions.error(cacheErrorEntry));
-        }
+        handleResponse(cacheEntry, api.dispatch, response?.status || 0);
```

Comment:
After learning how to type some of these internal RTK types, I pulled a lot of these boilerplate out.